### DISCUSSION TITLE
Any way to import custom popupMapping json?

### DISCUSSION THREAD

**Discussion by NePoCz** [florisboard/florisboard#2151] [Q&A] [answered]
Hi, thanks for the great keyboard, really like using it so far. Only thing I am missing is an option to import custom popupMapping jsons. I saw there is a ime/keyboards folder in app data, so should be possible (with root), but did not find any documentation. Is it similar to the .flex folders created for themes? Another option could be by manipulating the backup archive and then restoring it, which wouldnt need root. 
Or did I miss some easy way to do it?

**Comment by moonbeamcelery** [comment, marked as answer]
It's similar to .flex. Right now it's not possible to import keyboard .flex files, and the code only looks for the extension with a fixed name for keyboard mappings, and the extension with that name is included in the package. If you want custom popupMapping you have to compile the app, or try to change the extension within the apk.

**Reply by NePoCz** [reply]
Thanks for the reply. Are there plans to add the import feature in the near future? I know the full layout GUI editor is on the roadmap, but the json import could be much easier to implement and could be similar to the theme importer.

**Reply by moonbeamcelery** [reply]
Patrick wanted to add a language pack feature, which ideally include importing keyboard layouts and anything related to input methods for a language. But perhaps this would not be near future because he mentioned in the matrix chat that the priority is to get the next few alpha releases done.
